protocol/daemons/liquidation/client/sub_task_runner_test.go

```diff
@@ -0,0 +1,284 @@
+	"testing"
+)
+
+func TestGetAllSubaccounts(t *testing.T) {
+	df := flags.GetDefaultDaemonFlags()
+	tests := map[string]struct {
+		// mocks
+		setupMocks func(ctx context.Context, mck *mocks.QueryClient)
+
+		// expectations
+		expectedSubaccounts []satypes.Subaccount
+		expectedError       error
+	}{
+		"Success": {
+			setupMocks: func(ctx context.Context, mck *mocks.QueryClient) {
+				req := &satypes.QueryAllSubaccountRequest{
+					Pagination: &query.PageRequest{
+						Limit: df.Liquidation.SubaccountPageLimit,
+					},
+				}
+				response := &satypes.QuerySubaccountAllResponse{
+					Subaccount: []satypes.Subaccount{
+						constants.Carl_Num0_599USD,
+						constants.Dave_Num0_599USD,
+					},
+				}
+				mck.On("SubaccountAll", ctx, req).Return(response, nil)
+			},
+			expectedSubaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_599USD,
+				constants.Dave_Num0_599USD,
+			},
+		},
+		"Success Paginated": {
+			setupMocks: func(ctx context.Context, mck *mocks.QueryClient) {
+				req := &satypes.QueryAllSubaccountRequest{
+					Pagination: &query.PageRequest{
+						Limit: df.Liquidation.SubaccountPageLimit,
+					},
+				}
+				nextKey := []byte("next key")
+				response := &satypes.QuerySubaccountAllResponse{
+					Subaccount: []satypes.Subaccount{
+						constants.Carl_Num0_599USD,
+					},
+					Pagination: &query.PageResponse{
+						NextKey: nextKey,
+					},
+				}
+				mck.On("SubaccountAll", ctx, req).Return(response, nil)
+				req2 := &satypes.QueryAllSubaccountRequest{
+					Pagination: &query.PageRequest{
+						Key:   nextKey,
+						Limit: df.Liquidation.SubaccountPageLimit,
+					},
+				}
+				response2 := &satypes.QuerySubaccountAllResponse{
+					Subaccount: []satypes.Subaccount{
+						constants.Dave_Num0_599USD,
+					},
+				}
+				mck.On("SubaccountAll", ctx, req2).Return(response2, nil)
+			},
+			expectedSubaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_599USD,
+				constants.Dave_Num0_599USD,
+			},
+		},
+		"Errors are propagated": {
+			setupMocks: func(ctx context.Context, mck *mocks.QueryClient) {
+				req := &satypes.QueryAllSubaccountRequest{
+					Pagination: &query.PageRequest{
+						Limit: df.Liquidation.SubaccountPageLimit,
+					},
+				}
+				mck.On("SubaccountAll", ctx, req).Return(nil, errors.New("test error"))
+			},
+			expectedError: errors.New("test error"),
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			queryClientMock := &mocks.QueryClient{}
+			tc.setupMocks(grpc.Ctx, queryClientMock)
+
+			daemonClient := client.NewClient(log.NewNopLogger())
+			actual, err := client.GetAllSubaccounts(
+				daemonClient,
+				grpc.Ctx,
+				queryClientMock,
+				df.Liquidation.SubaccountPageLimit,
+			)
+			if err != nil {
+				require.EqualError(t, err, tc.expectedError.Error())
+				// The daemon initializes as unhealthy.
+				// If a request fails, the daemon will not be toggled to healthy.
+				require.Error(t, daemonClient.HealthCheck())
```

Comment:
The check here and 2 lines below is new - we validate daemon health based on whether or not the query returns an error.